### ISSUE TITLE
Set standard reverse proxy headers

### ISSUE THREAD

**Issue by anderspitman** [boringproxy/boringproxy#13]
Currently only sending `X-Forwarded-Host` upstream. Should probably send the others listed here:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling

May also want `X-Real-Ip` but I don't fully understand the difference between that and `X-Forwarded-For` yet.

**Comment by arp242** [comment]
> May also want X-Real-Ip but I don't fully understand the difference between that and X-Forwarded-For yet.

There isn't any really; other than having a different format. `X-Real-IP` always has just one IP address, whereas every proxy is expected to append to `X-Forwarded-For`.

The XFF is the most "standard" header, with XRI being used by comparatively few software packages AFAIK.

**Comment by anderspitman** [comment]
Awesome, thanks for the info :+1:

**Comment by anderspitman** [comment]
This is partially implemented in [#86535](https://github.com/boringproxy/boringproxy/commit/8653583cec7297314d1e6768f5172842afd75b7c), but there are issues with client TLS termination. See #114.

**Comment by anderspitman** [comment]
This should be resolved as of release v0.8.1.

**Comment by abulava** [comment]
@anderspitman I wonder why nobody asked you about [X-Forwarded-Proto](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) header for years.

`boringproxy` (version v0.9.1) is quite useless without it, to be honest.

I set it up with `TLS Termination: server` on `boringproxy` for [debugging](https://developers.facebook.com/tools/debug) Open Graph markup in a Rails app and that app renders
```
<meta content="http://example.com/" property="og:url">
<meta content="http://example.com/image.png" property="og:image">
```

Although the above links are reachable after redirects from http to https, Facebook bot fails to fetch `http://example.com/image.png` for some reason. The whole idea of using `boringproxy` is compromised, it can't replace [ngrok](ngrok.com) for me (**ngrok** sets `X-Forwarded-Proto`, those links start with "https://" and Facebook bot succeeds).
